docs/extras/use_cases/data_anonymization.ipynb

```diff
@@ -0,0 +1,476 @@
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from langchain.chains.transform import TransformChain\n",
+    "\n",
+    "anonymizer = PresidioAnonymizer()\n",
+    "\n",
+    "\n",
+    "def anonymize_func(inputs: dict) -> dict:\n",
+    "    text = inputs[\"text\"]\n",
+    "    return {\"output_text\": anonymizer.anonymize(text)}\n",
+    "\n",
+    "\n",
+    "anonymize_chain = TransformChain(\n",
+    "    input_variables=[\"text\"], output_variables=[\"output_text\"], transform=anonymize_func\n",
+    ")\n",
+    "\n",
+    "anonymize_chain(\"You can find our super secret data at https://supersecretdata.com\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\\\n",
+    "Later, you can, for example, use such anonymization as part of `SimpleSequentialChain`, as shown below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 40,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Set env var OPENAI_API_KEY or load from a .env file:\n",
+    "import dotenv\n",
+    "\n",
+    "dotenv.load_dotenv()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'input': 'You can find our super secret data at https://supersecretdata.com',\n",
+       " 'output': '\\nhttp://www.porter-hart.net/'}"
+      ]
+     },
+     "execution_count": 41,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from langchain.chains import SimpleSequentialChain\n",
+    "from langchain.prompts.prompt import PromptTemplate\n",
+    "from langchain.chains.llm import LLMChain\n",
+    "from langchain.llms.openai import OpenAI\n",
+    "\n",
+    "template = \"\"\"According to this text, where can you find our super secret data?\n",
+    "\n",
+    "{output_text}\n",
+    "\n",
+    "Answer:\"\"\"\n",
+    "prompt = PromptTemplate(input_variables=[\"output_text\"], template=template)\n",
+    "llm_chain = LLMChain(llm=OpenAI(), prompt=prompt)\n",
+    "\n",
+    "\n",
+    "sequential_chain = SimpleSequentialChain(chains=[anonymize_chain, llm_chain])\n",
+    "sequential_chain(\"You can find our super secret data at https://supersecretdata.com\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Future works\n",
+    "\n",
+    "- **deanonymization** - add the ability to reverse anonymization. For example, the workflow could look like this: `anonymize -> LLMChain -> deanonymize`. In this way, we will retain anonymity in requests to, for example, OpenAI, and then be able restore the original data.\n",
```

Comment:
By doing this, we will retain anonymity in requests to, for example, OpenAI, and then be able restore the original data.